src/page-multi-facility/FacilityInputForm.tsx

```diff
@@ -159,26 +160,43 @@ const FacilityInputForm: React.FC<Props> = ({ scenarioId }) => {
   const rejectionToast = useRejectionToast();
 
   const screenWidth = useScreenWidth();
-  const save = () => {
-    if (facilityName) {
-      // Set observedAt to right now when updating a facility from this input form
-      const modelUpdate = Object.assign({}, model[0]);
-      modelUpdate.observedAt = new Date();
 
+  const onSave = () => {
+    // Set observedAt to right now when updating a facility from this input form
+    const modelUpdate = Object.assign({}, model[0]) as ModelInputs;
```

Comment:
I'm not sure if this is the right thing to do here, but I needed it because the `model` type is `EpidemicModelState` and the `createOrUpdateFacility` is expecting a `Partial<Facility>` type, which sets its `modelInputs` property to a `ModelInputs` type. Is there another way to specify the right type here? I saw this works too but it's less precise maybe:

`const modelUpdate: any = Object.assign({}, model[0]);

Reply:
ah interesting .... ideally there would be some function to transform the model state to a `ModelInputs`; you would have to filter out the non-persistent keys and add `updatedAt` and `observedAt`. We do something similar already in the `saveFacility` function, which has been sufficient so far. I guess the danger here is that we know this assertion is false (it necessarily has extra fields and is missing required fields) so if the update object winds up sticking around in the context (which didn't really happen before, I guess) it could lead to some bugs. (I think it's handled safely in `createOrUpdateFacility` but it's still kind of an unexploded bomb)

Reply:
Okay thanks for the info, I tried to transform like in the `saveFacility` function and it looks like that works!